src/scales/scale.category.js

```diff
@@ -49,10 +49,20 @@ module.exports = Scale.extend({
 
 	getLabelForIndex: function(index, datasetIndex) {
 		var me = this;
-		var data = me.chart.data;
+		var chart = me.chart;
+		var data = chart.data;
 		var isHorizontal = me.isHorizontal();
 
-		if (data.yLabels && !isHorizontal) {
+		var ds = chart.getDatasetMeta(datasetIndex).controller;
+
+		// Do we have a getValuesScaleId function? (eg. bar chart)
```

Comment:
I feel like this comment could be improved a bit. "Do we have a getValuesScaleId function?" - I can see from the code that we're checking if this scale is the value scale. What I'm not sure of is what a value scale is or why it changes what label we return